All_years/04-Social_Care/Care_Home_data.R

```diff
@@ -0,0 +1,304 @@
+
+# read in demographic data
+sc_demog <- haven::read_sav(get_sc_demog_lookup_path())
+
+matched_ch_data <- ch_clean %>%
+  left_join(sc_demog, by = c("sending_location", "social_care_id"))
+
+
+# Data Cleaning Care Home Name and Postcode ---------------------------------------
+
+matched_ch_clean <- matched_ch_data %>%
+  # correct postcode formatting
+  mutate(across(contains("postcode"), .x = postcode(.x)))
+
+
+# postcode lookup
+postcode_lookup <- haven::read_sav(get_slf_postcode_path()) %>%
+  # select postcode
+  select(postcode) %>%
+  # format postcode
+  mutate(postcode = postcode(postcode))
+
+# CH lookup
+ch_lookup <- readxl::read_xlsx(get_slf_ch_path()) %>%
+  rename(
+    ch_postcode = "AccomPostCodeNo",
+    ch_name_lookup = "ServiceName"
+  ) %>%
+  select(
+    ch_postcode,
+    ch_name_lookup,
+    DateReg,
+    DateCanx
+  ) %>%
+  # format postcode
+  mutate(ch_postcode = postcode(ch_postcode))
+
+
+name_postcode_clean <- matched_ch_clean %>%
+  # deal with capitalisation of CH names
+  mutate(ch_name = stringr::str_to_title(ch_name)) %>%
+  # check postcodes and care home names are in lookup
+  filter(
+    !(ch_postcode %in% postcode_lookup),
+    !(ch_name %in% ch_lookup$ch_name_lookup)
+  ) %>%
+  ## ?? filter out if not in lookup OR make postcode/name NA if not in??
+  mutate(
+    ch_postcode = na_if(ch_postcode, ch_postcode %in% postcode_lookup),
+    ch_name = na_if(ch_name, ch_name %in% ch_lookup$ch_name_lookup)
+  ) %>%
+  arrange(ch_postcode) %>%
+  # where there is only a single Care Home at the Postcode (ever) just use that name, overwriting anything which was submitted
+  group_by(ch_postcode) %>%
+  # fill in names where NA but has a name in another record
+  fill(ch_name, .direction = "downup") %>%
+  # change name if CH name does not match previous name
+  mutate(ch_name_change_counter = pmax(ch_name != lag(ch_name) & !is.na(ch_name), FALSE, na.rm = TRUE)) %>%
+  mutate(ch_name = if_else(ch_name_change_counter == 1, last(ch_name), ch_name)) %>%
+  ungroup() %>%
+  # match CH names with CH lookup
+  left_join(ch_lookup, by = c("ch_postcode")) %>%
+  # check CH name matches the lookup name
+  mutate(name_match = if_else(ch_name != ch_name_lookup | is.na(ch_name_lookup), 0, 1)) %>%
+  # replace ch name with lookup name if not a match
+  mutate(ch_name = if_else(name_match == 0 & !is.na(ch_name_lookup), ch_name_lookup, ch_name)) %>%
+  # check admission date within CH dates
+  mutate(date_check = if_else(is.na(DateReg) | is.na(DateCanx) | ch_admission_date %in% range(DateReg, DateCanx), TRUE, FALSE))
+  ## ?? filter out the FALSE? where admission date is not within the CH dates??
+
+
+# Data Cleaning Care Home Data ---------------------------------------
+
+ch_data_clean <- matched_ch_data %>%
+  # sort data
+  arrange(sending_location, social_care_id, ch_admission_date, period) %>%
+  group_by(sending_location, social_care_id, ch_admission_date, period) %>%
```

Comment:
This sort won't do anything since you're grouping on the same values. Actually, you shouldn't be grouping by `period` here as we're interested in fixing the values for the 'whole episode'. Did I group by period in the SPSS?
```suggestion
  arrange(sending_location, social_care_id, ch_admission_date, period) %>%
  group_by(sending_location, social_care_id, ch_admission_date) %>%
```